cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -877,6 +878,8 @@ public void close() {
   protected RetryableProvisionException handleApiException(ApiException e) throws RetryableProvisionException {
```

Comment:
Not part of your changes. But can you add handleApiException call in getDataprocCluster as well, instead of the same check i.e. `e.getStatusCode().getCode().getHttpStatusCode() / 100 != 4`